cosmosdb/common/ps-account-update-region.ps1

```diff
@@ -11,28 +11,18 @@
 # Variables - ***** SUBSTITUTE YOUR VALUES *****
 $resourceGroupName = "myResourceGroup" # Resource Group must already exist
 $accountName = "myaccount" # Must be all lower case
-$locations = @("West US", "East US") # Regions ordered by failover priority
+$locations = @("East US", "West US") # Regions ordered by failover priority
 # --------------------------------------------------
 
-# Get existing Cosmos DB account
-# $account = Get-AzCosmosDBAccount -ResourceGroupName $resourceGroupName -Name $accountName
+Write-Host "Get Cosmos DB account"
+$account = Get-AzCosmosDBAccount -ResourceGroupName $resourceGroupName -Name $accountName
 
-# Eventually transition to Update-AzCosmosDBAccountRegion with -Location or -LocationObject
-# Update-AzCosmosDBAccountRegion -InputObject $account -Location $locations
-# ForEach ($name in $locations){ $locationObjects += New-AzCosmosDBLocationObject -LocationName $name -FailoverPriority ($i++) }
-# Update-AzCosmosDBAccountRegion -InputObject $account -LocationObject $locationObjects
-
-# Use Set-AzResource with property object pending transition to Update-AzCosmosDBAccountRegion
-$resourceType = "Microsoft.DocumentDb/databaseAccounts"
-$apiVersion = "2020-03-01"
-$locationObjects = @()
 $i = 0
-ForEach ($location in $locations) { $locationObjects += @{ locationName = "$location"; failoverPriority = $i++ } }
-$CosmosDBProperties = @{
-    databaseAccountOfferType = "Standard";
-    locations = $locationObjects;
+$locationObjects = @()
+
+ForEach ($location in $locations) {
+    $locationObjects += New-AzCosmosDBLocationObject -LocationName $location -FailoverPriority ($i++)
 }
 
-Set-AzResource -ResourceGroupName $resourceGroupName -ResourceType $resourceType `
-    -ApiVersion $apiVersion -Name $accountName `
-    -PropertyObject $CosmosDBProperties -Force
+Write-Host "Update Cosmos DB account region(s)"
+Update-AzCosmosDBAccountRegion -InputObject $account -LocationObject $locationObjects
```

Comment:
Can a user update a region and specify it is AZ enabled?

Reply:
Yes, assuming the region object-based approach is used as in ps-account-update-region.ps1. 

However, there is not at this time a convenient way to look up programmatically if a region is AZ-enabled; user must consult https://azure.microsoft.com/global-infrastructure/regions/ or similar resource to determine whether the region(s) in question is/are AZ-enabled (there is feedback about this on docs and github in various places, e.g. https://github.com/MicrosoftDocs/feedback/issues/544).

Once user determines if a region is AZ-enabled, then would need to so denote by adding the -IsZoneRedundant $true/$false parameter to New-AzCosmosDBLocationObject.

I'll extend the sample to use a hashtable of (region, AZ-enabled) and build the location list that way to show the approach.

Reply:
There is also this approach: run the following Powershell and look for any SKU showing enumerated AZs. This region - westus - does not show as AZ-enabled at this time, and this is borne out by none of the returned SKUs having an AZ list.
Get-AzComputeResourceSku | where {$_.Locations.Contains("westus")}

Reply:
OK, pushed the extended sample with AZ designation.